stups-spring-oauth2-server/src/main/java/org/zalando/stups/oauth2/spring/authorization/HttpUserRolesProvider.java

```diff
@@ -0,0 +1,100 @@
+package org.zalando.stups.oauth2.spring.authorization;
+
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+
+import static org.springframework.security.oauth2.common.OAuth2AccessToken.BEARER_TYPE;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.core.ParameterizedTypeReference;
+
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+
+import org.springframework.util.Assert;
+
+import org.springframework.web.client.RestTemplate;
+
+import org.zalando.stups.oauth2.spring.server.TokenResponseErrorHandler;
+import org.zalando.stups.spring.http.client.ClientHttpRequestFactorySelector;
+
+public class HttpUserRolesProvider implements UserRolesProvider {
+
+    private final String roleInfoUri;
+
+    private final String rolePrefix;
+
+    private static final String ROLE_SEPARATOR = "_";
+
+    public static final String EMPLOYEES = "/employees";
+
+    public static final String DEFAULT_ROLE = "ROLE_USER";
+
+    RestTemplate restTemplate = buildRestTemplate();
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    public HttpUserRolesProvider(final String roleInfoUri, final String rolePrefix) {
```

Comment:
A little bit more info would be great. What is roleInfoUri. I also think (I may be wrong) that roleInfoUrl is tightly coupled to the internal Zalando infra... (that's zack - right?)).